### PR TITLE
deployment-service: try to create admin token using iam v2 if v1 fails

### PR DESCRIPTION
**Pull request by srenatus**
Also:
* diagnostics/context: don't swallow that error

Co-authored-by: Blake Johnson <bstier@chef.io>
Signed-off-by: Stephan Renatus <srenatus@chef.io>

### CODE REVIEW

File: components/automate-cli/pkg/diagnostics/context.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (c *testContext) adminToken() (string, error) {
 		})
 
 		if err != nil {
-			return "", nil
+			return "", err
```
Comment (stevendanna) [inline components/automate-cli/pkg/diagnostics/context.go]: 🎉

File: components/automate-deployment/pkg/server/api_token.go [hunk 2]
```diff
@@ -71,6 +75,20 @@ func generateAdminToken(ctx context.Context,
 		Subjects: []string{fmt.Sprintf("token:%s", response.Id)},
 		Resource: "*",
 	})
+	if isUseV2Error(err) {
+		_, err = authzV2Client.CreatePolicy(ctx, &authz_v2.CreatePolicyReq{
+			Id:   "diagnostics-admin-token",
```
[the remark this answers is not in the document]
Comment (srenatus) [reply components/automate-deployment/pkg/server/api_token.go]: I suppose the behaviour of this for non-ci use is a bit worrying -- the v1 code will just create a new token (and not clean it up) every time it's run, the v2 code will probably fail the second time it's run...

[the remark this answers is not in the document]
Comment (srenatus) [reply components/automate-deployment/pkg/server/api_token.go]: Happy to pick up follow-up tasks if we deem them useful; I'll merge this to unblock the other PR 😃

[the remark this answers is not in the document]
Comment (blakestier) [reply components/automate-deployment/pkg/server/api_token.go]: Alternately, we can add the token to the admin policy (doesn't fix the clean up issue, but would prevent 💣 )

[the remark this answers is not in the document]
Comment (srenatus) [reply components/automate-deployment/pkg/server/api_token.go]: Yes, indeed. Is that what we do for `chef-automate iam token create --admin`?

**Pull request merged**